### PR TITLE
fix(settings): Add public/private email options in the profile page

### PR DESCRIPTION
**Pull request by rohitkrai03**
Added an option for users to choose between public or private email. JS logic with defined functionality implemented.

fixes https://github.com/openshiftio/openshift.io/issues/1775

Includes HTML/CSS code from https://github.com/fabric8-ui/fabric8-ui/pull/2461

### CODE REVIEW

**Comment by sbose78** [comment]
[test]

**Comment by sbose78** [comment]
@rohitkrai03 could you please amend the commit message to the "fix(..): ....." format ?
So that irrespective of whether we do a `sqash and merge` we can do a release with this commit.

**Comment by rohitkrai03** [comment]
@sbose78 Somehow i couldn't find any test specific to update profile component. Can you point me to the file so I could write some tests for emailPrivate functionality? 
Will change the commit message to proper format. Thanks.

**Comment by sbose78** [comment]
@rohitkrai03 There aren't any yet . Would you be able to contribute a couple of tests related to this area? Please follow the official documentation for tests.
Also, here's a nice PR by @bartoszmajsak https://github.com/fabric8-ui/fabric8-ui/pull/2234 which you can use as a reference.

**Comment by joshuawilson** [comment]
[test]

**Comment by rohitkrai03** [comment]
@joshuawilson CI build is failing because [ngx-login-client](https://github.com/fabric8-ui/ngx-login-client) doesn't have changes in Profile type related to emailPrivate. I can go ahead and create another PR for that.

**Comment by joshuawilson** [comment]
yes, let's include that first

**Comment by rohitkrai03** [comment]
[test]

**Comment by sbose78** [comment]
@rohitkrai03 please update your branch so that you have the latest `ngx-login-client` here, unless you've already pulled it in.

[test]

**Comment by rohitkrai03** [comment]
@sbose78 branch is now updated with latest pull from master.

File: src/app/profile/update/update.component.html [hunk 1]
```diff
@@ -17,13 +17,30 @@ <h1>Edit Profile</h1>
...
-          <div [ngClass]="{'has-error': emailInvalid}">
+          <label for="email" class="control-label required-pf email-alignment-label">Email</label>
+          <div [ngClass]="{'has-error': emailInvalid}" class="email-alignment">
             <input type="email" class="form-control" id="email" name="email" #_email
                     placeholder="name@gmail.com"
                     [maxLength]="255"
                     [(ngModel)]="email"
                     (change)="validateEmail()">
+            <div class="radio email-alignment_settings">
+              <label>
```
[the remark this answers is not in the document]
Comment (rohitkrai03) [reply src/app/profile/update/update.component.html]: I actually referred to this PR - https://github.com/fabric8-ui/fabric8-ui/pull/2461/files for HTML/CSS code and assumed it was done for some alignment issue. I'll change it and update the PR.

[the remark this answers is not in the document]
Comment (mindreeper2420) [reply src/app/profile/update/update.component.html]: Wrapping the `<input>` in a `<label>` is per the Bootstrap docs: https://getbootstrap.com/docs/3.3/css/#forms-example. Scroll down to where is shows
```
<div class="checkbox">
  <label>
    <input type="checkbox"> Check me out
  </label>
</div>
```

[the remark this answers is not in the document]
Comment (rohitkrai03) [reply src/app/profile/update/update.component.html]: @mindreeper2420 Thanks for the clarification.

[the remark this answers is not in the document]
Comment (joshuawilson) [reply src/app/profile/update/update.component.html]: thanks @mindreeper2420 just looks wrong to me, but I will let it go

**Comment by joshuawilson** [comment]
[test]

**Comment by alexeykazakov** [comment]
[test]

**Pull request merged**